### PR TITLE
Research Software Ecosystem (RSEc) and the Galaxy Codex - blog post

### PR DESCRIPTION
**Pull request by paulzierep**
Research Software Ecosystem (RSEc) needs a bog post for the ELIXIR report.
Here is the initial draft, including the Galaxy Codex as well.
@bebatut @bgruening @hmenager @supernord

Please let me know if there is anything particular you would like to see here, will try to finalize beginning of next week.

### CODE REVIEW

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+15. Conda id
+16. Conda version
+
+The data is stored as TSV and JSON in the CoDex repository.
+
+### RSEc integration
+
+The output of the CoDex was integrated into the RSEc by the development of a [Galaxy Tool Metadata GitHub Action](https://github.com/research-software-ecosystem/utils/tree/main/galaxytool-import),
+which populates the RSEc entries with metadata from the CoDex.
+An example of a tool that has metadata from multiple sources, including Galaxy is [abricate](https://github.com/research-software-ecosystem/content/tree/c840b13928330e97c7eb6ccb6a779ff9b0f37835/data/abricate).
```
Comment (bgruening) [inline content/news/2024-07-11-RSEc-CoDex/index.md:84]: Image of this?

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    8. Getting usage statistics from usegalaxy.eu
+- Create an interactive table for all tools: [All tools](https://galaxyproject.github.io/galaxy_tool_metadata_extractor/)
+- Filter the tool suite per community
+- Create an interactive table for all registered communities, e.g. [microGalaxy](https://galaxyproject.github.io/galaxy_tool_metadata_extractor/microgalaxy/)
+
+<img src="./Preprint_flowchart.png" style="max-width: 400px" alt="CoDex" />
+
+### Tool metadata output
+
+The script will generate a TSV file with each tool found in the list of GitHub repositories and metadata for these tools:
```
Comment (bgruening) [inline content/news/2024-07-11-RSEc-CoDex/index.md:59]: Link to the TSV file?

**Comment by bgruening** [comment]
Cool, thanks @paulzierep

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+## ELIXIR Research Software Ecosystem (RSEc)
+
+<img src="./rsec.svg" style="max-width: 400px" alt="RSEc" />
+
+The [ELIXIR Research Software Ecosystem (RSEc)](https://research-software-ecosystem.github.io/index) is supported and coordinated by [ELIXIR Europe](https://elixir-europe.org/).
+It makes use of a sophisticated CI architecture that collects metadata for research software metadata, providing a unified resource for researchers. 
+
+The RSEc collects metadata from [bio.tools](https://bio.tools), [Bioconda](https://bioconda.github.io/), BioContainers, [Debian Med](https://www.debian.org/devel/debian-med/index.de.html), [Galaxy CoDex](https://github.com/galaxyproject/galaxy_codex),
+[OpenEBench](https://openebench.bsc.es), and [WorkflowHub](https://workflowhub.eu/) available in the [Content repositroy](https://github.com/research-software-ecosystem/content).
```
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:20]:
```suggestion
[OpenEBench](https://openebench.bsc.es), and [WorkflowHub](https://workflowhub.eu/) available in the [Content repository](https://github.com/research-software-ecosystem/content).
```

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+More information about the sources can be found [here](https://research-software-ecosystem.github.io/sources).
+
+The CI architecture is [openly available](https://github.com/research-software-ecosystem/utils) and can be adapted and leveraged to integrate other metadata sources as well.
+Recently, the architecture was extended to include Galaxy Codex's metadata.
```
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:24]:
```suggestion
The architecture was recently extended to include metadata from the Galaxy Codex.
```

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
+---
+title: 'The ELIXIR Research Software Ecosystem (RSEc) and the Galaxy Codex - powerful GitHub Bot architectures to collect research software metadata'
```
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:2]:
```suggestion
title: 'Enabling architectures that collect community-centric research software metadata'
```

Comment (supernord) [reply content/news/2024-07-11-RSEc-CoDex/index.md:2]: Just a suggestion

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+---
+
+# The ELIXIR Research Software Ecosystem (RSEc) and the Galaxy Codex - powerful GitHub Bot architectures to collect research software metadata
+
+## ELIXIR Research Software Ecosystem (RSEc)
+
+<img src="./rsec.svg" style="max-width: 400px" alt="RSEc" />
+
+The [ELIXIR Research Software Ecosystem (RSEc)](https://research-software-ecosystem.github.io/index) is supported and coordinated by [ELIXIR Europe](https://elixir-europe.org/).
+It makes use of a sophisticated CI architecture that collects metadata for research software metadata, providing a unified resource for researchers.
```
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:17]:
```suggestion
It makes use of a sophisticated CI architecture that collects metadata for research software metadata, providing a unified resource for researchers, developers and curators. 
```

File: content/news/2024-07-11-RSEc-CoDex/index.md (hunk 3 above)
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:20]:
```suggestion
[OpenEBench](https://openebench.bsc.es), and [WorkflowHub](https://workflowhub.eu/). This metadata is made available in the [RSEc content repositroy](https://github.com/research-software-ecosystem/content).
```

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+The RSEc collects metadata from [bio.tools](https://bio.tools), [Bioconda](https://bioconda.github.io/), BioContainers, [Debian Med](https://www.debian.org/devel/debian-med/index.de.html), [Galaxy CoDex](https://github.com/galaxyproject/galaxy_codex),
+[OpenEBench](https://openebench.bsc.es), and [WorkflowHub](https://workflowhub.eu/) available in the [Content repositroy](https://github.com/research-software-ecosystem/content).
+More information about the sources can be found [here](https://research-software-ecosystem.github.io/sources).
+
+The CI architecture is [openly available](https://github.com/research-software-ecosystem/utils) and can be adapted and leveraged to integrate other metadata sources as well.
+Recently, the architecture was extended to include Galaxy Codex's metadata.
+
+The CI bots perform the following tasks weekly:
+
+* Downloads dumps of all meta sources into the content repository
```
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:28]:
```suggestion
* Download dumps of all metadata sources into the content repository
```

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+* Stores the metadata entries of each tool from all sources into a dedicated folder in the content repository
```
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:29]:
```suggestion
* Store the metadata entries for each tool, from all sources, in a dedicated folder in the content repository
```

File: content/news/2024-07-11-RSEc-CoDex/index.md (hunk 4 above)
Comment (supernord) [reply content/news/2024-07-11-RSEc-CoDex/index.md:24]: This content could be moved to the `Galaxy Codex` section below.

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+- Extract all tools by
+    1. Parsing tool GitHub repository from [Planemo monitor listed](https://github.com/galaxyproject/planemo-monitor)
+    2. Checking in each repo, their `.shed.yaml` file and filter for categories, such as metagenomics 
+    3. Extracting metadata from the `.shed.yaml`
+    4. Extracting the `requirements` from the Galaxy tool metadata (XML files) to get the exact version supported in Galaxy
+    5. Checking available against conda version
+    6. Extracting bio.tools information from the Galaxy tool metadata (XML files) if available
+    7. Checking the availability of the tool on one of the 3 main Galaxy instances (usegalaxy.eu, usegalaxy.org, usegalaxy.org.au)
+    8. Getting usage statistics from usegalaxy.eu
+- Create an interactive table for all tools: [All tools](https://galaxyproject.github.io/galaxy_tool_metadata_extractor/)
```
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:51]:
```suggestion
- Create an interactive table for all tools: [All tools](https://galaxyproject.github.io/galaxy_codex/)
```

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+- Filter the tool suite per community
+- Create an interactive table for all registered communities, e.g. [microGalaxy](https://galaxyproject.github.io/galaxy_tool_metadata_extractor/microgalaxy/)
```
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:53]:
```suggestion
- Create a separate interactive table for all registered communities, e.g. [microGalaxy](https://galaxyproject.github.io/galaxy_tool_metadata_extractor/microgalaxy/)
```

File: content/news/2024-07-11-RSEc-CoDex/index.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+<img src="./CI_01.png" style="max-width: 400px" alt="CI" />
+
+## The Galaxy Codex
+
+[Galaxy Communities Dock aka Galaxy Codex](https://github.com/galaxyproject/galaxy_codex) is a catalog of Galaxy resources (tools, training, workflows) that can be filtered for any community.
+In this blog post, we only cover the tool aspect of the catalog since this is used as the source for the contribution of Galaxy to the RSEc.
+
+### Tool metadata collection
+
+A GitHub action performs every week the following steps:
```
Comment (supernord) [inline content/news/2024-07-11-RSEc-CoDex/index.md:40]:
```suggestion
A GitHub action performs the following steps every week:
```

**Comment by bgruening** [comment]
Cool, thanks!

**Pull request merged**